f/frizzle/comapeo/comapeo_observations.py

```diff
@@ -92,20 +108,50 @@ def fetch_comapeo_projects(
             if project["project_id"] not in comapeo_project_blocklist
         ]
 
-    if not comapeo_projects:
-        raise ValueError("No valid projects found.")
-
     return comapeo_projects
 
 
+@retry(
+    stop_max_attempt_number=3,
+    wait_exponential_multiplier=500,
+)
```

Comment:
I forget - if this fails 3 times in the row, what happens?  Will the entire script fail?  Or does it silently move on?

Reply:
No. What happens is what I was describing in "Screenshots" in the PR body.

The script moves on after 3 failed retries, but tracks the failure by setting `attachment_failed` = true. The script then throws a RuntimeError at the end if that's the case, which therefore marks the run as Failed.

Reply:
Thank you. Ok, that part makes sense to me now -- and so my confusion is actually something different.

Can you confirm: Is it actually retrying?  I'd claim it's not.

I thought `@retry` would retry if it catches an exception, but you never give it the opportunity to.  Line 132 catches & silences any exception that might have come from this function.

Reply:
You're right, it's not. It seemed like it did when I ran it with some attachments I knew would fail, but I just confirmed with logging that it only runs once.

In that case, I'm doubting whether `@retry` is the solution here. Because if I don't catch & silence the exception, it will cause the script to fail (and not proceed) right away after 3 attempts.

...I'm also increasingly sympathetic to your line of reasoning [here](https://github.com/ConservationMetrics/frizzle/issues/90):

> This is ridiculous that we even retry. All this code is polling code, so even if it fails it will retry again after the poll interval. I'm going to just rip out the retries instead of dealing with all the desired behavior up above.

I'm tempted to drop the retry and just keep the `attachment_failed` bool.

Reply:
I'd be ok with that.